.github/workflows/pytest.yml

```
@@ -0,0 +1,16 @@
+name: PyTest
+on: pull_request
```

The CI should also run after a PR was merged to `main`.
In addition, usually, you want to run the CI only for pull requests with target set to `main` branch.
The last option allows you to manually trigger the workflow which can come in handy

```suggestion
on:
  push:
    branches: [main]
  pull_request:
    branches: [main]
  workflow_dispatch:
```